packages/client/src/execution/vmexecution.ts

```diff
@@ -989,8 +989,36 @@ export class VMExecution extends Execution {
    */
   async executeBlocks(first: number, last: number, txHashes: string[]) {
     this.config.logger.info('Preparing for block execution (debug mode, no services started)...')
-    const vm = await this.vm.shallowCopy(false)
 
+    let vm
+    const block = await this.vm.blockchain.getBlock(first)
+    const parentBlock = await this.vm.blockchain.getBlock(block.header.parentHash)
+    const startExecutionParentTd = await this.chain.getTd(block.hash(), parentBlock.header.number)
+
+    const startExecutionHardfork = this.config.execCommon.getHardforkBy({
+      blockNumber: block.header.number,
+      td: startExecutionParentTd,
+      timestamp: block.header.timestamp,
+    })
+
+    // Setup VM with verkle state manager if Osaka is active
+    if (
+      this.config.execCommon.hardforkGteHardfork(startExecutionHardfork, Hardfork.Osaka) &&
+      this.config.statelessVerkle
+    ) {
+      const verkleSM = await StatelessVerkleStateManager.create({
+        common: this.vm.common,
+        verkleCrypto: (this.vm.stateManager as StatelessVerkleStateManager).verkleCrypto,
+      })
+      verkleSM.initVerkleExecutionWitness(parentBlock.header.number, parentBlock.executionWitness!)
```

Comment:
we don't need to call this with parentBlock as curren't blocks witnesses are inited and used in runBlock

Reply:
So should I call it with `block.executionWitness` instead of `parentBlock.executionWitness`?  It errors out with ` ERROR Error starting client startExecutionHardfork is not defined` if I don't call that `initVerkleExecutionWitness` function before instantiating the VM.